### ISSUE TITLE
Reduce the dependencies on different versions of `rand`

### ISSUE THREAD

**Issue by jvff** [ZcashFoundation/zebra#2391]
## Motivation

Currently, Zebra depends on 3 different versions of the `rand` crate:

- ~`rand v0.4.7` is used by the `temp-dir v0.3.7` dev-dependency of `zebra-test`~
- `rand v0.7.3` is used:
  - directly by `zebra-consensus`
- `rand v0.8.4` is used:
  - directly by `tower-batch` as a dev-dependency
  - directly by `zebra-network` and `zebra-consensus`
  - directly by `zebra-test`

This increases the size of the libraries and binaries, increases build time, and creates conflicts between different struct versions when developing.

### Out of Scope

- `rand v0.7.3` is used:
  - indirectly by the `sentry v0.21.0` dependency of `zebrad`
  - indirectly by the `proptest v0.10.1` dependency of `zebra-test`
- `rand v0.8.4` is used:
  - indirectly by the `halo2` dependency of `zebra-chain`

## Tasks

1. Use a single `rand` version in Zebra's direct dependencies

## Related issues

This is blocked by #2985.

**Comment by mpguerra** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/2391?planning-poker) with ZenHub @conradoplg @dconnolly @jvff @oxarbitrage @teor2345 @upbqdn

**Comment by jvff** [comment]
This seems to only depend on `x25519-dalek` having its 2.0 version released. So this should be fixed by upgrading to that version once it is released. I'll close this issue for now, and we can create a new issue for the update later.
